mmv1/third_party/terraform/services/storagetransfer/resource_storage_transfer_job.go.erb

```diff
@@ -394,12 +395,70 @@ func transferOptionsSchema() *schema.Schema {
 					ValidateFunc: validation.StringInSlice([]string{"DIFFERENT", "NEVER", "ALWAYS"}, false),
 					Description:  `When to overwrite objects that already exist in the sink. If not set, overwrite behavior is determined by overwriteObjectsAlreadyExistingInSink.`,
 				},
+				"metadata_options": metadataOptionsSchema(),
 			},
 		},
 		Description: `Characteristics of how to treat files from datasource and sink during job. If the option delete_objects_unique_in_sink is true, object conditions based on objects' last_modification_time are ignored and do not exclude objects in a data source or a data sink.`,
 	}
 }
 
+func metadataOptionsSchema() *schema.Schema {
+return &schema.Schema{
+Type:     schema.TypeList,
+Optional: true,
+MaxItems: 1,
+Elem: &schema.Resource{
+Schema: map[string]*schema.Schema{
+"symlink": {
+Type:         schema.TypeString,
+Optional:     true,
+ValidateFunc: validation.StringInSlice([]string{"SYMLINK_UNSPECIFIED","SYMLINK_SKIP","SYMLINK_PRESERVE"}, false),
```

Comment:
More and more we've been weighing leaving enum fields as just normal strings without any validation. This removes the benefit of plan-time validation/error checking for users, but would allow the provider to immediately support all values the API may support in the future.

As a future user yourself, do you find yourself caring more about the validation or immediate support for future values in these fields?

Reply:
I totally see your point about immediate support in the context of this project and I think this would be my preference for this library. Though in other and personal projects I'd probably vote for strong validation. Also depending on the tech stack and how hard it would be to automate downstream changes.